File: app/src/main/java/wangdaye/com/geometricweather/data/api/CNWeatherApi.java
package wangdaye.com.geometricweather.data.api;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import wangdaye.com.geometricweather.data.entity.result.cn.CNWeatherResult;

/**
 * CN weather api.
 * */

public interface CNWeatherApi {

    @GET("v4/{city_id}.json")
    Call<CNWeatherResult> getWeather(@Path("city_id") String city_id);
}

File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/CNWeatherApi.kt
package wangdaye.com.geometricweather.weather.apis

import retrofit2.http.GET
import retrofit2.http.Path
import wangdaye.com.geometricweather.weather.json.cn.CNWeatherResult

/**
 * CN weather api.
 */
interface CNWeatherApi {

    @GET("v4/{city_id}.json")
    suspend fun getWeather(@Path("city_id") city_id: String): CNWeatherResult
}